core/src/Dscp/Crypto/Util.hs

```diff
@@ -31,4 +31,4 @@ secretFromSeed input = asum
         return $ withIntSeed (fromIntegral seed) genSecretKey
     , do
         return $ withSeed input genSecretKey
-    ]
+    ] ?: error "impossible"
```

Comment:
Frankly, why not rewrite it as `maybe (withSeed input genSecretKey) $ ...` without this `error`?